legacy-fabric-registry-sync-api-v2/common/src/main/java/net/legacyfabric/fabric/api/registry/v2/registry/holder/Registry.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2020 - 2024 Legacy Fabric
+ * Copyright (c) 2016 - 2022 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.legacyfabric.fabric.api.registry.v2.registry.holder;
+
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import net.legacyfabric.fabric.api.event.Event;
+import net.legacyfabric.fabric.api.registry.v2.event.RegistryBeforeAddCallback;
+import net.legacyfabric.fabric.api.registry.v2.event.RegistryEntryAddedCallback;
+import net.legacyfabric.fabric.api.util.Identifier;
+
+public interface Registry<T> extends Iterable<T> {
```

Comment:
Registry -> RegistryExtensions? I don't think its a good idea to keep it as the same name as an existing minecraft class

Reply:
`RegistryExtensions` doesn't fit in this case as this interface can be used to implement a custom registry implementation.
However maybe we should prefix those classes with either `Fabric`, `LF` or `LegacyFabric` to differentiate them from vanilla classes as you've said.

Reply:
Renamed to `FabricRegistry`.